server/models/commentModel.js

```diff
@@ -0,0 +1,16 @@
+import mongoose from 'mongoose';
+import autoIncrementId from './autoIncrementModel';
+
+const CommentSchema = new mongoose.Schema({
+  description: { type: String, required: true },
+  createdBy: { type: String, required: true },
+
+}, { timestamps: { createdAt: 'createdAt' } });
+
+async function prepProcess(next) {
+  const increment = autoIncrementId.bind(this, 'commentId', 'id');
```

Comment:
why binding?

Reply:
this is util function common for all which gives autoincremented id. Internally it uses 'pre' save of mongoose which requires this of current schema.

Reply:
autoIncrementedId is a util function common for all to get incremented unique id for a particular schema. Internally it uses 'pre' save of mongoose which required 'this' object of the current schema. that's why binding is used.

Reply:
why not using default mongo objectId?

Reply:
We are using both objetid and autoincremented id. we put autoincrement id in an intention to use in client side as the object-id is too long to use and pass it in params